rust/src/sip/sip.rs

```diff
@@ -112,27 +109,31 @@ impl SIPState {
         }
     }
 
+    fn build_tx_request(&mut self, input: &[u8], request: Request) {
+        let mut tx = self.new_tx(crate::core::Direction::ToServer);
+        tx.request = Some(request);
+        if let Ok((_, req_line)) = sip_take_line(input) {
+            tx.request_line = req_line;
+        }
+        self.transactions.push(tx);
+    }
+
     // app-layer-frame-documentation tag start: parse_request
     fn parse_request(&mut self, flow: *const core::Flow, stream_slice: StreamSlice) -> bool {
         let input = stream_slice.as_slice();
-        let _pdu = Frame::new(
-            flow,
-            &stream_slice,
-            input,
-            input.len() as i64,
-            SIPFrameType::Pdu as u8,
-        );
-        SCLogDebug!("ts: pdu {:?}", _pdu);
 
         match sip_parse_request(input) {
             Ok((_, request)) => {
+                let _pdu = Frame::new(
```

Comment:
frame should not depend on the parser returning ok

Reply:
see above